src/compose/service.ts

```diff
@@ -1124,26 +1147,50 @@ export class Service {
 	): ServiceConfig['volumes'] {
 		let volumes: ServiceConfig['volumes'] = [];
 
-		_.each(composeVolumes, (volume) => {
-			const isBind = _.includes(volume, ':');
-			if (isBind) {
-				const [bindSource, bindDest, mode] = volume.split(':');
-				if (!path.isAbsolute(bindSource)) {
-					// namespace our volumes by appId
-					let volumeDef = `${appId}_${bindSource.trim()}:${bindDest.trim()}`;
+		// namespace our volumes by appId
+		const namespaceVolume = (volumeSource: string) =>
+			`${appId}_${volumeSource.trim()}`;
+
+		for (const volume of composeVolumes || []) {
+			const isAnonymousOrTmpfs =
+				(LongSyntaxVolume.is(volume) && volume.type === 'tmpfs') ||
+				(typeof volume === 'string' && !volume.includes(':'));
+
+			let isBindMount = false;
+
+			if (isAnonymousOrTmpfs) {
+				// Push anonymous volume. Anonymous volumes can only be
+				// set by using the `VOLUME` instruction inside a Dockerfile.
+				// Tmpfs volumes do not require namespacing by appId.
+				volumes.push(volume);
+			} else if (LongSyntaxVolume.is(volume)) {
+				// Handle long syntax volumes of type 'tmpfs' and 'volume'
+				isBindMount = path.isAbsolute(volume.source!);
+				if (!isBindMount) {
+					volume.source = namespaceVolume(volume.source!);
```

Comment:
What happens if you assign a source for a `tmpfs` volume? Same as above, if volume is `tmpfs` this will assign as source `namespaceVolume(undefined)` resulting in `<appId>_undefined`, if I'm understanding correctly (which I'm not sure I am)

Reply:
If you assign `source` to a long syntax tmpfs volume, docker-compose will error (and therefore balena-compose should error), so we shouldn't need to handle this. Would you prefer the check to be here though?

Reply:
On the new code, there is already a check in this branch for the mount being a volume or bind, so it should not be necessary to add the check here.